tracer/src/Datadog.Trace/AppSec/EventTrackingSdk.cs

```diff
@@ -17,172 +18,185 @@ namespace Datadog.Trace.AppSec;
         }
-
-        /// <summary>
-        /// Sets the details of a custom event the local root span
-        /// </summary>
-        /// <param name="eventName">the name of the event to be tracked</param>
-        /// <param name="metadata">Metadata associated with the custom event</param>
-        [PublicApi]
-        public static void TrackCustomEvent(string eventName, IDictionary<string, string> metadata)
+    }
+
+    /// <summary>
+    /// Sets the details of a custom event the local root span
+    /// </summary>
+    /// <param name="eventName">the name of the event to be tracked</param>
+    [PublicApi]
+    public static void TrackCustomEvent(string eventName)
+    {
+        TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackCustomEvent);
+        TrackCustomEvent(eventName, null, Tracer.Instance);
+    }
+
+    /// <summary>
+    /// Sets the details of a custom event the local root span
+    /// </summary>
+    /// <param name="eventName">the name of the event to be tracked</param>
+    /// <param name="metadata">Metadata associated with the custom event</param>
+    [PublicApi]
+    public static void TrackCustomEvent(string eventName, IDictionary<string, string> metadata)
+    {
+        TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackCustomEvent_Metadata);
+        TrackCustomEvent(eventName, metadata, Tracer.Instance);
+    }
+
+    internal static void TrackCustomEvent(string eventName, IDictionary<string, string> metadata, Tracer tracer)
+    {
+        if (string.IsNullOrEmpty(eventName))
         {
-            TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackCustomEvent_Metadata);
-            TrackCustomEvent(eventName, metadata, Tracer.Instance);
+            ThrowHelper.ThrowArgumentNullException(nameof(eventName));
         }
 
-        internal static void TrackCustomEvent(string eventName, IDictionary<string, string> metadata, Tracer tracer)
-        {
-            if (string.IsNullOrEmpty(eventName))
-            {
-                ThrowHelper.ThrowArgumentNullException(nameof(eventName));
-            }
+        var span = tracer?.ActiveScope?.Span;
 
-            var span = tracer?.ActiveScope?.Span;
-
-            if (span is null)
-            {
-                ThrowHelper.ThrowException("Can't create a tracking event with no active span");
-            }
+        if (span is null)
+        {
+            ThrowHelper.ThrowException("Can't create a tracking event with no active span");
+        }
 
-            var setTag = TaggingUtils.GetSpanSetter(span);
+        var setTag = TaggingUtils.GetSpanSetter(span);
 
-            setTag(Tags.AppSec.Track(eventName), "true");
+        setTag(Tags.AppSec.Track(eventName), "true");
 
-            setTag($"_dd.{Tags.AppSec.Events}{eventName}.sdk", "true");
+        setTag($"_dd.{Tags.AppSec.Events}{eventName}.sdk", "true");
 
-            if (metadata != null)
+        if (metadata != null)
```

Comment:
Same as suggested earlier in with LoginEvent.Failure

Reply:
good idea, changed in [5eb2354](https://github.com/DataDog/dd-trace-dotnet/pull/6225/commits/5eb235421602d965bfe3b2ceb925b15fac34cad7)